### PR TITLE
Add computed custom fields

### PR DESCRIPTION
**Pull request by rco-odoo**
Compute method is given by a snippet in a text field.

### CODE REVIEW

**Comment by rco-odoo** [comment]
@KangOl can you review this for me? Thanks

File: openerp/addons/base/ir/ir_model.py [hunk 1]
```diff
@@ -319,6 +327,12 @@ def _onchange_related(self):
                 return {'warning': {'title': _("Warning"), 'message': e.message}}
             self.ttype = field.type
             self.relation = field.comodel_name
+            self.readonly = True
```
Comment (KangOl) [inline openerp/addons/base/ir/ir_model.py]: Maybe this line should be in its own commit.

File: openerp/models.py [hunk 2]
```diff
@@ -65,6 +66,19 @@
 
 AUTOINIT_RECALCULATE_STORED_FIELDS = 1000
 
+# base environment for doing a safe eval
+SAFE_EVAL_BASE = {
+    'time': time,
+    'datetime': datetime,
+    'dateutil': dateutil,
```
Comment (KangOl) [inline openerp/models.py]: Add `math` module

**Comment by KangOl** [comment]
:+1:
